RingierAG/react-native-read-more-text
# Fix unhandled promise rejection

Fix unhandled promise rejection (TypeError: Cannot read property 'measure' of null).
It happens because componentWillUnmount sometimes is invoked before the async componentDidMount finishes (usually before the 2nd measureHeightAsync function is invoked) and at this point, this._text reference is undefined.

**How to test this:**
* Switch to:
 "react-native-read-more-text": "git+ssh://git@github.com/RingierAG/react-native-read-more-text.git#fix/unhandled-promise-rejection",
in package.json in newBlick repo (native-components).
* Run npm install (before this, delete the react-native-read-more-text dependency from node_modules)
* Start the Gallery on native (you can use: /nb-mock/catalog/widget/gallery).
* Go to the Gallery with 30 slides
* Scroll fast a couple of slides and then click on a slide (fullscreen will open)
* Scroll fast a couple of slides in fullscreen mode and then click the Close button
* There should be no error (red screen), Gallery should function normally.
* Repeat this for a few times. :)
